### PR TITLE
[quant] making quant utilities inplace

### PR DESCRIPTION
**Pull request by z-a-f**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#25054 [quant] making quant utilities inplace**

Differential Revision: [D16974198](https://our.internmc.facebook.com/intern/diff/D16974198)

### CODE REVIEW

File: torch/quantization/quantize.py [hunk 1]
```diff
@@ -240,15 +239,11 @@ def quantize(model, run_fn, run_args, mapping=DEFAULT_MODULE_MAPPING):
         run_fn: a function for evaluating the prepared model, can be a
             function that simply runs the prepared model or a training loop
         run_args: positional arguments for `run_fn`
-
-    Return:
-        A quantized model
     """
     model.eval()
```
Comment (jerryzh168) [inline torch/quantization/quantize.py]: can we do a deepcopy of model before this line

File: torch/quantization/quantize.py [hunk 2]
```diff
@@ -260,21 +255,18 @@ def quantize_dynamic(model, qconfig_dict=DEFAULT_QCONFIG_DICT, mapping=DEFAULT_D
...
-    model = prepare(model)
-    model = convert(model, DEFAULT_QAT_MODULE_MAPPING)
-    return model
+    prepare(model)
+    convert(model, DEFAULT_QAT_MODULE_MAPPING)
 
 def quantize_qat(model, run_fn, run_args):
     r"""Do quantization aware training and output a quantized model
     """
     model.train()
```
Comment (jerryzh168) [inline torch/quantization/quantize.py]: same for this

File: torch/quantization/quantize.py [hunk 3]
```diff
@@ -283,26 +275,21 @@ def convert(module, mapping=DEFAULT_MODULE_MAPPING):
...
+                         nni.LinearReLU,
+                         nni.ConvReLU2d)
 
     for name, mod in module.named_children():
-        new_mod = convert(mod, mapping)
-        if new_mod is not mod:
-            reassign[name] = new_mod
+        if type(mod) not in SWAPPABLE_MODULES:
+            convert(mod, mapping)
+        swapped_mod = swap_module(mod, mapping)
```
Comment (jerryzh168) [inline torch/quantization/quantize.py]: nit: merge these two lines into one

File: torch/quantization/quantize.py (hunk 1 above)
Comment (zafartahirov) [reply torch/quantization/quantize.py]: Wait, didn't we decide not to deepcopy?

File: torch/quantization/quantize.py [hunk 4]
```diff
@@ -240,12 +241,11 @@ def quantize(model, run_fn, run_args, mapping=DEFAULT_MODULE_MAPPING):
         run_fn: a function for evaluating the prepared model, can be a
             function that simply runs the prepared model or a training loop
         run_args: positional arguments for `run_fn`
-
-    Return:
```
Comment (jerryzh168) [inline torch/quantization/quantize.py]: this still returns a quantized model

**Review by jerryzh168 — APPROVED** [review]
please fix comments

**Pull request closed without merging**